### PR TITLE
1450049: Replace `-` with `_` in completion functions

### PR DESCRIPTION
**Pull request by kahowell**
Also added a test that will execute each script with `bash --posix`, so
if we regress, it should show up in unit test failures.

### CODE REVIEW

**Comment by kahowell** [comment]
b6d1310..d811021: forgot to add script

**Comment by kahowell** [comment]
d811021..73f5c92: fix spacing issue in test

File: etc-conf/rhn-migrate-classic-to-rhsm.completion.sh [hunk 1]
```diff
@@ -24,4 +24,4 @@ _rhn-migrate-classic-to-rhsm()
 	return 0
 }
 
-complete -F _rhn-migrate-classic-to-rhsm rhn-migrate-classic-to-rhsm
+complete -F _rhn-migrate_classic_to_rhsm rhn-migrate-classic-to-rhsm
```
Comment (wottop) [inline etc-conf/rhn-migrate-classic-to-rhsm.completion.sh]: "_rhn-migrate_classic_to_rhsm" shouldn't it be an underscore between rhn and migrate?

Comment (kahowell) [inline etc-conf/rhn-migrate-classic-to-rhsm.completion.sh]: Nice catch! Fixed in 80ec7aa.

**Comment by kahowell** [comment]
73f5c92..80ec7aa: fix an instance where I missed a '-'

**Comment by wottop** [comment]
ack

**Pull request merged**